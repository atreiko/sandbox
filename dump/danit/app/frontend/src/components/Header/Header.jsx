import React from 'react'
import PropTypes from 'prop-types'
import { CustomLink } from '../CustomLink'


const Header = ({ user }) => {
    return (
        <header>
            <div>{user.name}</div>
            <div>{user.age}</div>
            <div>{user.email}</div>

            <CustomLink to='/'>Home</CustomLink>
            <CustomLink to='/favorites'>Favorites</CustomLink>
            <CustomLink to='/sent'>Sent</CustomLink>
            <CustomLink to='/inbox'>Inbox</CustomLink>
        </header>
    )
}

Header.propTypes = {
    user: PropTypes.object.isRequired
}

export default Header
