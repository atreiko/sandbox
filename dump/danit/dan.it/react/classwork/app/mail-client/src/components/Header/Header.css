.header__title {
  color: red;
}
/*# sourceMappingURL=Header.css.map */