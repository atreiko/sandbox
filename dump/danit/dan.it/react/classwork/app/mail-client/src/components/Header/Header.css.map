{
    "version": 3,
    "mappings": "AACG,AAAD,cAAQ,CAAC;EACP,KAAK,EAAE,GAAG;CACX",
    "sources": [
        "Header.scss"
    ],
    "names": [],
    "file": "Header.css"
}