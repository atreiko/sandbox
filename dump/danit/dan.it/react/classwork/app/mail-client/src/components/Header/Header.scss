.header {
  &__title {
    color: red;
  }
}