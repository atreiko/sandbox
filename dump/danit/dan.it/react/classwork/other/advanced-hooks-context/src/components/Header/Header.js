import React, { memo, useContext } from 'react'
import ThemeContext from '../../../../../practice/swapi-context/src/context/themeContext'
import Button from '../Button/Button'

function Header() {
  console.log('Rendering Header.js')
  const context = useContext(ThemeContext);

  return (
    <div>
      <h2>Header</h2>
      <Button title='Toggle theme' onClick={context.toggleTheme} />
    </div>
  )
}

export default memo(Header)
