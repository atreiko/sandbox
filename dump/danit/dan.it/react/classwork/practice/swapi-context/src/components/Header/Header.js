import React, { useContext } from 'react'
import ThemeContext from '../../context/themeContext';

function Header({ user, setUser }) {
  const themeContext = useContext(ThemeContext);
  const isAuth = !!user

  const logOutUser = () => {
    setUser(null);
  }

  const selectOptions = Object.keys(themeContext.themes).map(o => (
    <option key={o} value={o}>{o}</option>
  ))

  const changeTheme = (e) => {
    themeContext.setTheme(themeContext.themes[e.target.value])
  }

  return (
    <div>
      {!isAuth && <div>Welcome, anonimous</div>}
      {isAuth && <div>{user.login}</div>}
      {isAuth && <div><button onClick={logOutUser}>Log out</button></div>}
      <select onChange={changeTheme} defaultValue={themeContext.default}>
        {selectOptions}
      </select>
    </div>
  )
}

export default Header
