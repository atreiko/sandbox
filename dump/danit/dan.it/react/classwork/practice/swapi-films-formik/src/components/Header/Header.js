import React from 'react'

function Header({user, setUser}) {
  const isAuth = !!user
  
  const logOutUser = () => {
    setUser(null);
  }

  return (
    <div>
      {!isAuth && <div>Welcome, anonimous</div>}
      {isAuth && <div>{user.login}</div>}
      {isAuth && <div><button onClick={logOutUser}>Log out</button></div>}
    </div>
  )
}

export default Header
