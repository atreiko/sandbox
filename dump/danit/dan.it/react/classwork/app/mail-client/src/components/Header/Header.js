import React from 'react';
import './Header.scss';
import PropTypes from 'prop-types';
import { getUserSelector } from '../../store/user/selectors'
import { connect } from 'react-redux';

export const Header = (props) => {
  const { title, user } = props;

  return (
    <div className='header'>
      <h2 className='header__title'>Header</h2>
      <div id='header-title' data-testid='testTitle'>{title}</div>
      {!!user && <div id='user-login'>{user.login}</div>}
    </div>
  );
}

Header.propTypes = {
  title: PropTypes.string,
  user: PropTypes.object
}

Header.defaultProps = {
  title: 'Default title'
}

const mapStateToProps = (state) => {
  return {
    user: getUserSelector(state)
  }
}

export default connect(mapStateToProps)(Header);