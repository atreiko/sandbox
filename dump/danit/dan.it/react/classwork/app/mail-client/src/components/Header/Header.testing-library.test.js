import { Header } from "./Header";
import { render } from "@testing-library/react";

describe("Testing Header.js", () => {
  test("Smoke test of Header", () => {
    render(<Header />);
  });

  test("Header shows title from props", () => {
    const testTitle = "TEST TITLE";
    const { /*container,*/ getByText, getByTestId } = render(
      <Header title={testTitle} />
    );

    const title = getByText(testTitle);
    expect(title).toBeInTheDocument();
    // getByTestId('testTitle')
  });

  test("User login div is not shown when user is not passed in props", () => {
    render(<Header />);

    const login = document.getElementById("user-login");
    expect(login).toBe(null);
  });

  test("User login is shown when user is passed in props", () => {
    const testUser = { login: "TEST_LOGIN" };
    const { getByText } = render(<Header user={testUser} />);
    getByText(testUser.login);
  });
});

describe("Snapshot testing Header.js", () => {
  test("Simple header render", () => {
    const { container } = render(<Header />);
    expect(container.innerHTML).toMatchSnapshot();
  });

  test("Rendering header with title", () => {
    const { container } = render(<Header title="TEST_TITLE" />);
    expect(container.innerHTML).toMatchSnapshot();
  });

  test("Rendering header with user", () => {
    const { container } = render(<Header user={{ login: "TEST_USER" }} />);
    expect(container.innerHTML).toMatchSnapshot();
  });

  test("Inline snapshot of header", () => {
    const { container } = render(<Header />);
    expect(container.innerHTML).toMatchInlineSnapshot(
      `"<div class=\\"header\\"><h2 class=\\"header__title\\">Header - class component!</h2><div id=\\"header-title\\" data-testid=\\"testTitle\\">Default title</div></div>"`
    );
  });
});
