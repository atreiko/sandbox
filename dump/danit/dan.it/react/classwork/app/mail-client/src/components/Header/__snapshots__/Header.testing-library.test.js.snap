// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing Header.js Rendering header with title 1`] = `"<div class=\\"header\\"><h2 class=\\"header__title\\">Header - class component!</h2><div id=\\"header-title\\" data-testid=\\"testTitle\\">TEST_TITLE</div></div>"`;

exports[`Snapshot testing Header.js Rendering header with user 1`] = `"<div class=\\"header\\"><h2 class=\\"header__title\\">Header - class component!</h2><div id=\\"header-title\\" data-testid=\\"testTitle\\">Default title</div><div id=\\"user-login\\">TEST_USER</div></div>"`;

exports[`Snapshot testing Header.js Simple header render 1`] = `"<div class=\\"header\\"><h2 class=\\"header__title\\">Header - class component!</h2><div id=\\"header-title\\" data-testid=\\"testTitle\\">Default title</div></div>"`;
